src/Signer/Key/InMemory.php

```diff
@@ -16,15 +17,24 @@ final class InMemory implements Key
     private string $contents;
     private string $passphrase;
 
+    /** @param non-empty-string $contents */
     private function __construct(string $contents, string $passphrase)
     {
+        if ($contents === '') {
+            throw InvalidKeyProvided::cannotBeEmpty();
+        }
+
         $this->contents   = $contents;
         $this->passphrase = $passphrase;
     }
 
     public static function empty(): self
```

Comment:
Should we mark this as `@deprecated`?

Reply:
I don't think so: legit usages exist.
What matters is that consumer is aware it's empty.